realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -187,6 +241,11 @@ void RealSenseNodeFactory::setUpResinChuck()
         bool found = false;
         while (!found)
         {
+            if (ros::isShuttingDown())
+            {
+                ROS_ERROR_STREAM("ROS is shutting down returning early");
```

Comment:
Any log message that does not trigger the actual failure should be debug or info level logging.